Userland/Applications/PixelPaint/VectorscopeWidget.cpp

```diff
@@ -0,0 +1,138 @@
+            auto u_index = u_v_to_index(yuv.u);
+            auto v_index = u_v_to_index(yuv.v);
+            m_vectorscope_data[u_index][v_index]++;
+        }
+    }
+
+    auto maximum = full_bitmap->width() * full_bitmap->height() * pixel_percentage_for_max_brightness * pixel_percentage_for_max_brightness;
+
+    for (size_t i = 0; i < m_vectorscope_data.size(); ++i) {
+        for (size_t j = 0; j < m_vectorscope_data[i].size(); ++j) {
+            m_vectorscope_data[i][j] = sqrt(m_vectorscope_data[i][j]) / maximum;
+        }
+    }
+
+    return {};
+}
+
+void VectorscopeWidget::rebuild_vectorscope_image()
+{
+    m_vectorscope_image = MUST(Gfx::Bitmap::try_create(Gfx::BitmapFormat::BGRA8888, size()));
+    m_vectorscope_image->fill(Color::Transparent);
+
+    Gfx::Painter base_painter(*m_vectorscope_image);
+    Gfx::AntiAliasingPainter painter(base_painter);
+
+    auto const scope_size = min(height(), width());
+    auto const color_vector_size = max(1.0, static_cast<double>(scope_size) / static_cast<double>(u_v_steps));
+    auto const size_1x1 = Gfx::FloatSize { 2.5f, 2.5f } * static_cast<float>(color_vector_size);
+
+    base_painter.translate(width() / 2, height() / 2);
+    painter.translate(static_cast<float>(width()) / 2.0f, static_cast<float>(height()) / 2.0f);
+    for (size_t u_index = 0; u_index < u_v_steps; ++u_index) {
+        for (size_t v_index = 0; v_index < u_v_steps; ++v_index) {
+            auto const color_vector = ColorVector::from_indices(u_index, v_index);
+            auto const brightness = m_vectorscope_data[u_index][v_index];
+            if (brightness < 0.0001)
+                continue;
+            auto const pseudo_rect = Gfx::FloatRect::centered_at(color_vector.to_vector(scope_size) * 2.0, size_1x1);
+            auto color = Color::from_yuv(0.6, color_vector.u, color_vector.v);
+            color.saturate_to(1.0 - min(brightness, 1.0));
+            color.set_alpha(static_cast<u8>(min(sqrt(brightness), 1.0) * NumericLimits<u8>::max()));
+            base_painter.fill_rect(pseudo_rect.to_rounded<int>(), color);
+        }
+    }
+}
+
+void VectorscopeWidget::paint_event(GUI::PaintEvent& event)
+{
+    GUI::Painter base_painter(*this);
+    Gfx::AntiAliasingPainter painter(base_painter);
+    base_painter.add_clip_rect(event.rect());
+    // From this point on we're working with 0,0 as the scope center to make things easier.
+    base_painter.translate(width() / 2, height() / 2);
+    painter.translate(static_cast<float>(width()) / 2.0f, static_cast<float>(height()) / 2.0f);
+
+    auto const graticule_color = Color::White;
+    auto const scope_size = min(height(), width());
+    auto const graticule_size = scope_size / 6;
+    auto const graticule_thickness = graticule_size / 12;
+    auto const entire_scope_rect = Gfx::FloatRect::centered_at({ 0, 0 }, { scope_size, scope_size });
+
+    painter.fill_ellipse(entire_scope_rect.to_rounded<int>(), Color::Black);
```

Comment:
```suggestion
        painter.fill_ellipse(entire_scope_rect.to_rounded<int>().shrunken(graticule_thickness*2, graticule_thickness*2), Color::Black);
```
You should shrink the inner ellipse here, otherwise it clashes with the graticule (border :^)), below, adding some aliasing.

Reply:
(this is why there's a faint black grey-ish border around the edge currently)